helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/constraints/SoftConstraint.java

```diff
@@ -57,15 +58,19 @@
     _minScore = minScore;
   }
 
+  float getMaxScore() {
+    return _maxScore;
+  }
+
+  float getMinScore() {
+    return _minScore;
+  }
+
   /**
-   * The scoring function returns a score between MINIMAL_SCORE and MAXIMUM_SCORE, which is then
-   * weighted by the
-   * individual normalized constraint weights.
-   * Each individual constraint will define the meaning of MINIMAL_SCORE to MAXIMUM_SCORE
-   * differently.
-   * @return float value representing the score
+   * Evaluate and give a score for an potential assignment partition -> instance
+   * @return The score of the assignment in float value
    */
-  abstract float getAssignmentOriginScore(AssignableNode node, AssignableReplica replica,
+  abstract float evaluateAssignment(AssignableNode node, AssignableReplica replica,
```

Comment:
We are still using the normalize method you proposed earlier, right? I think we need a method name to differentiate this score from the normalized score. "evaluateAssignment" is not clear enough for me.